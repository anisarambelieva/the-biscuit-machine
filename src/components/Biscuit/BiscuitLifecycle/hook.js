import { useEffect, useState } from "react";

const useBiscuitLifecycle = (setBakedBiscuitsCount, bakedBiscuitsCount) => {
  const [stamped, setStamped] = useState(false);
  const [baked, setBaked] = useState(false);

  useEffect(() => {
    if (baked) {
      setBakedBiscuitsCount((bakedBiscuitsCount) => bakedBiscuitsCount + 1);
    }
  }, [baked, setBakedBiscuitsCount]);

  return {
    stamped,
    setStamped,
    baked,
    setBaked,
  };
};

export default useBiscuitLifecycle;
