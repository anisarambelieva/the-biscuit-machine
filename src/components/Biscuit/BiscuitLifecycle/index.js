import { useEffect, useState } from "react";
import { Col } from "react-bootstrap";

import BakedBiscuit from "../BakedBiscuit";
import Dough from "../Dough";
import Biscuit from "../RawBiscuit";

const BiscuitLifecycle = ({
  animationState,
  onStamp,
  machineState,
  setBakedBiscuitsCount,
  bakedBiscuitsCount,
}) => {
  const [stamped, setStamped] = useState(false);
  const [baked, setBaked] = useState(false);

  useEffect(() => {
    if (baked) {
      setBakedBiscuitsCount((bakedBiscuitsCount) => bakedBiscuitsCount + 1);
    }
  }, [baked, setBakedBiscuitsCount]);

  return (
    <>
      <Col md="1">
        <Dough
          onAnimationEnd={setStamped}
          onStamp={onStamp}
          state={animationState}
          machineState={machineState}
        />
      </Col>

      <Col md="1">
        {stamped && (
          <Biscuit
            onAnimationEnd={setBaked}
            state={animationState}
            onStamp={setStamped}
          />
        )}
      </Col>

      <Col md={{ span: 1, offset: 2 }}>
        {baked && <BakedBiscuit state={animationState} onBaked={setBaked} />}
      </Col>
    </>
  );
};

export default BiscuitLifecycle;
