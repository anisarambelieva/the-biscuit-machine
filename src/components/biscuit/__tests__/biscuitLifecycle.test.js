import { render } from "@testing-library/react";

import BiscuitLifecycle from "../biscuitLifecycle.js";

describe("<BiscuitLifecycle />", () => {
  it("should match snapshot", () => {
    const { container } = render(<BiscuitLifecycle />);

    expect(container).toMatchSnapshot();
  });
});
