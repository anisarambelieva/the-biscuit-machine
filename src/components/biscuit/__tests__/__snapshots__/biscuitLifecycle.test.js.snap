// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BiscuitLifecycle /> should match snapshot 1`] = `
<div>
  <div
    class="col-md-1"
  >
    <div
      class="dough"
      style="margin: 0px; position: absolute; padding-left: 20px;"
    >
      Dough
    </div>
  </div>
  <div
    class="col-md-1"
  />
  <div
    class="col-md-1 offset-md-2"
  />
</div>
`;
